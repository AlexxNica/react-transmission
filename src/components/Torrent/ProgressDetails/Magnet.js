import React from 'react';
import { FormattedMessage, injectIntl, defineMessages } from 'react-intl';

const messages = defineMessages({
  needs: {
    id: 'torrent.progress.magnet.needs',
    defaultMessage: 'needs',
  },
  retrieving: {
    id: 'torrent.progress.magnet.retrieving',
    defaultMessage: 'retrieving',
  },
});

function ProgressMagnet({ torrent, intl }) {
  const metadataStatus = torrent.isStopped ? messages.needs : messages.retrieving;

  return (
    <FormattedMessage
      id='torrent.progress.magnet'
      defaultMessage={`Magnetized transfer - {metadataStatus} metadata ({completed})`}
      values={{
        metadataStatus: intl.formatMessage(metadataStatus),
        completed: intl.formatNumber(torrent.metadataPercentComplete, {style: 'percent'}),
      }}
    />
  );
}

export default injectIntl(ProgressMagnet);
