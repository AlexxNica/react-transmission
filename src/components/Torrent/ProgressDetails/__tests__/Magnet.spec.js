import React from 'react';
import createComponentWithIntl from 'test/util/createComponentWithIntl';

import Magnet from '../Magnet';

test('Magnet', () => {
  const component = createComponentWithIntl(
    <Magnet torrent={{metadataPercentComplete: 0.75}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Magnet stopped', () => {
  const component = createComponentWithIntl(
    <Magnet torrent={{metadataPercentComplete: 0.75, isStopped: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
