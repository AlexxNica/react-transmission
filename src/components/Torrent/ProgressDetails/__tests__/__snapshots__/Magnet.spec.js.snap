exports[`test Magnet 1`] = `
<span>
  Magnetized transfer - retrieving metadata (75%)
</span>
`;

exports[`test Magnet stopped 1`] = `
<span>
  Magnetized transfer - needs metadata (75%)
</span>
`;
